tests/testthat/test_grad.R

```diff
@@ -15,19 +15,87 @@ test_that("linear model's analytical gradient is close to numerical one", {
   n_obs <- 32; n_pred <- 4
```

Comment:
You'll probably want to rename this file since it's testing more than the gradient now.